src/components/select/select-list/select-list.component.js

```diff
@@ -93,63 +107,92 @@ const SelectList = React.forwardRef(
       [anchorElement]
     );
 
-    const optionRefList = useMemo(
-      () =>
-        React.Children.map(children, (child) => {
-          if (child?.type === Option || child?.type === OptionRow) {
-            return React.createRef();
-          }
-          return null;
-        }).filter((child) => child),
-      [children]
-    );
-
     const handleSelect = useCallback(
       (optionData) => {
         onSelect({ ...optionData, selectionType: "click" });
       },
       [onSelect]
     );
 
-    const childIds = useMemo(() => React.Children.map(children, () => guid()), [
+    const childIdsRef = useRef(null);
+
+    // childIds should be stable except when children are added or removed - can't use useMemo
+    // as that isn't absolutely guaranteed to never rerun when dependencies haven't changed.
+    const setChildIds = () => {
+      const idArray = [];
+      for (let i = 0; i < children.length; i++) {
+        idArray.push(guid());
+      }
+      childIdsRef.current = idArray;
+    };
+
+    if (
+      childIdsRef.current === null ||
+      childIdsRef.current.length !== React.Children.count(children)
+    ) {
+      setChildIds();
+    }
+
+    const childIds = childIdsRef.current;
+
+    const childrenList = useMemo(() => React.Children.toArray(children), [
       children,
     ]);
 
-    const childrenWithListProps = useMemo(
-      () =>
-        React.Children.map(children, (child, index) => {
-          if (!child || (child.type !== Option && child.type !== OptionRow)) {
-            return child;
-          }
+    const optionChildrenList = useMemo(
```

Comment:
there's certainly a string argument for using `displayName` instead of `type` but in the absence of either a specific bug raised for this on Select or a general refactor across all components I'd rather leave this as it was in this PR.

Which yes means including `OptionRow`, I'm not sure how I missed that but Maciek already caught it (above).